jetty-util/src/main/java/org/eclipse/jetty/util/thread/ReservedThreadExecutor.java

```diff
@@ -212,52 +202,61 @@ public boolean tryExecute(Runnable task)
     {
         if (LOG.isDebugEnabled())
             LOG.debug("{} tryExecute {}", this, task);
-
         if (task == null)
             return false;
 
-        ReservedThread thread = _stack.pollFirst();
-        if (thread == null)
+        boolean offered = _queue.offer(task);
```

Comment:
I don't agree. I don't think we want to start a thread unless we are size 0, even if the offer fails.

Reply:
More precisely, if we fail in the offer here, it is not because of insufficient size.  We have reserved thread(s), they've just not arrived yet. starting another is not helpful.

Reply:
resolving to discuss below